docker-compose.yaml

```diff
@@ -1,18 +1,34 @@
-version: "3"
+version: "3.8"
 services:
   server:
     build:
       context: .
-      dockerfile: Dockerfile.server
+      dockerfile: Dockerfile.cfnts
     depends_on:
       - memcache
+    volumes:
+      - ./tests:/tests
+      - ./scripts:/scripts
+    entrypoint: ["/scripts/run_server.sh"]
   client:
     build:
       context: .
-      dockerfile: Dockerfile.client
+      dockerfile: Dockerfile.cfnts
     depends_on:
       - server
+    volumes:
+      - ./tests:/tests
+      - ./scripts:/scripts
+    entrypoint: ["/scripts/run_client.sh"]
   memcache:
     build:
       context: .
       dockerfile: Dockerfile.memcache
+    volumes:
+      - ./scripts:/scripts
+    entrypoint: ["/scripts/run_memcached.sh"]
+  clippy:
+    build:
+      context: .
+      dockerfile: Dockerfile.clippy
+    command: ["--all-features", "--", "-D", "warnings"]
```

Comment:
Are there unit tests in this repo to run as well?

Did you consider running cargo test && cargo clippy via GitHub's native CI stuff? @armfazh helped me set up https://github.com/cloudflare/daphne/blob/main/.github/workflows/daphneci.yml

Reply:
Nice! I'll give that a try. There are some unit tests, but coverage is far from complete.